### DISCUSSION TITLE
"Wrong" grafana dashboard loaded

### DISCUSSION THREAD

**Discussion by nezvocowa** [Security-Onion-Solutions/securityonion#4219] [Unsupported Versions] [answered]
- 2.3.50 (problem occurred on 2.3.40)
- Installed from SecOnion ISO 2.3.40 and upgraded with 2.3.50 ISO
- Airgap
- Heavy (physical), Manager (VM)
- so-status OK (on every node and master)
- state.highstate OK (on every node and master)

Everything is running and working except that the dashboards for my Heavy Nodes are `Search Nodes`, not `Sensor Nodes` as expected by previous versions of SO.
Therefore I can't see Zeek, Suricata, Steno and pcap related info that the `Sensor Nodes` dashboard show.

This behavior wasn't noticed in previous installations <= 2.3.30 and I haven't got time to try and do a fresh install of 2.3.50 with the same configuration.

Perhaps a specific dashboard would be the solution for `Heavy Nodes` in the future that includes both "elastic and pcap" info e.g combining a `"Sensor Search" Nodes` dashboard?

**TLDR**: Is there a way to swap/change what kind of dashboard in Grafana that's associated to my Heavy Nodes without reinstalling? I haven't figured out where in the salt config to actually make the change 🤔

**Comment by m0duspwnens** [comment, marked as answer]
@nezvocowa 
 There is definitely a need for a dedicated Grafana dashboard for heavy nodes.

 Can you check `/opt/so/saltstack/local/pillar/data/` `nodetab.sls` and `sensortab.sls` and confirm that there is data for your heavy node? If there is, can you change the guid in either the `nodetab.sls` or the `sensortab.sls` so that they don't match? Once that is done, as root run `salt-call state.apply grafana queue=True`. Does that allow you to see both the sensor and search dashboards in Grafana for your heavy node now?

 If it doesn't, you can remove the data for the heavy node from the appropriate sls file under `/opt/so/saltstack/local/pillar/data/`, then delete its respective dashboard under `/opt/so/conf/grafana/grafana_dashboards` `search_nodes` or `sensor_nodes`. Rerun `salt-call state.apply grafana queue=True` to finalize the changes.

**Reply by nezvocowa** [reply]
@m0duspwnens 
I can't get them both to work at the same time, and that's fine for me right now!
If I find time, I'll try to create a hybrid of both dashboards for my heavy nodes.

**For others**:
If I change the `guid` in `sensortab.sls` and run `salt-call state.apply grafana queue=True`, the sensor dashboard shows up and vice versa. But not both at the same time 😃👍  

Thank you for the help!
